sdk/resource/resource_test.go

```diff
@@ -769,3 +770,30 @@ func TestWithContainer(t *testing.T) {
 		string(semconv.ContainerIDKey): fakeContainerID,
 	}, toMap(res))
 }
+
+func TestResourceRace(t *testing.T) {
+	// Creating Resources should also be free of any ASAN issues,
+	// because Resources are immutable.
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	for i := 0; i < 2; i++ {
```

Comment:
```suggestion
	for i := 0; i < 2; i++ {
		wg.Add(1)
```

this would be safer if one would like to change the number of goroutines in the test